File: Behavioral/Memento/Python/history.py
# history.py

class History:
  def __init__(self):
    self._history = []

  def save(self, memento):
    if memento:
      self._history.append(memento)

  def undo(self):
    if len(self._history) > 1:
      self._history.pop()  # Remove the latest state
    return self._history[-1] if self._history else None

File: Behavioral/Memento/JavaScript/history.js
// history.js

class History {
  constructor() {
    this._history = [];
  }

  save(memento) {
    if (memento) {
      this._history.push(memento);
    }
  }

  undo() {
    if (this._history.length > 1) {
      this._history.pop(); // Remove the latest state
    }
    return this._history.length > 0 ? this._history[this._history.length - 1] : null;
  }
}

module.exports = History;
